2021/day10/day10.java

```diff
@@ -0,0 +1,83 @@
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.LinkedList;
+
+public class day10 {
+    public static ArrayList<ArrayList<Character>> readInput(String filename) {
+        ArrayList<ArrayList<Character>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Character> line = new ArrayList<>();
+                String s = sc.next();
+                for (int i = 0; i < s.length(); i++) {
+                    line.add(s.charAt(i));
+                }
+                input.add(line);
+            }
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    public static void syntaxScoring() {
+        ArrayList<ArrayList<Character>> input = readInput("data.txt");
+        ArrayList<Long> autocompleteScores = new ArrayList<>();
+        int syntaxScore = 0;
+
+        for (ArrayList<Character> line : input) {
+
+            LinkedList<Character> stack = new LinkedList<>();
+            Boolean isCorrupted = false;
+            long autoTmpScore = 0;
+
+            for (int i = 0; i < line.size(); i++) {
+                if (line.get(i) == '(' || line.get(i) == '<' ||
+                    line.get(i) == '[' || line.get(i) == '{')
+                    stack.addFirst(line.get(i));
+
+                else if (stack.peek() == line.get(i) - 2 || stack.peek() == line.get(i) - 1)
+                    stack.removeFirst();
+
+                else {  // mark line as corrupted and add to syntax score
+                    isCorrupted = true;
+                    if (line.get(i) == ')')
+                        syntaxScore += 3;
+                    else if (line.get(i) == ']')
+                        syntaxScore += 57;
+                    else if (line.get(i) == '}')
+                        syntaxScore += 1197;
+                    else if (line.get(i) == '>')
+                        syntaxScore += 25137;
+                    break;
+                }
+            }  
+            if (isCorrupted == false) {
+                for (Character element : stack) {
+                    if (element == '(')
+                        autoTmpScore = autoTmpScore * 5 + 1;
+                    else if (element == '[')
+                        autoTmpScore = autoTmpScore * 5 + 2;
+                    else if (element == '{')
+                        autoTmpScore = autoTmpScore * 5 + 3;
+                    else if (element == '<')
+                        autoTmpScore = autoTmpScore * 5 + 4;
+                }
+                autocompleteScores.add(autoTmpScore);
+            }
+        }
+        System.out.println(syntaxScore);
+        Collections.sort(autocompleteScores);
+        System.out.println(autocompleteScores.get(autocompleteScores.size() / 2));
```

Comment:
I see your problem with print vs return now. In here I guess printing is easier and the only way to return both is to create a separate class. However, you can also extract the common parts into a common function and then just modify the parts that are specific to each part. It may not be easy for this example, so I don't deem it necessary for a merge :) TL;DR: either
1. Make a class with two values and return it (easy), or
2. Encapsulate the common code into a common function and make part1/part2 minimalistic